### PR TITLE
[2/2] Support BOSH v256+

### PR DESCRIPTION
**Pull request by adamstegman**
https://www.pivotaltracker.com/story/show/120141267

BOSH v256+ enforces that each job can only see its own properties. This means we can't cheat and peek into other jobs' configuration to share credentials and quorum settings. Add some new properties to work around this.

Update the specs with the latest version of BOSH, using bosh-lite instead of reaching out to AWS.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# -*- mode: ruby -*-
```
Comment (jeremiemv) [inline Vagrantfile]: We always need need Vagrant for Bosh-list, now the config is explicit?

Comment (adamstegman) [inline Vagrantfile]: Not sure I follow—this is auto-generated when you run `vagrant init cloudfoundry/bosh-lite`, checking it in means you don't have to run that yourself to run the tests.

File: spec/support/bosh.rb [hunk 2]
```diff
@@ -48,30 +43,60 @@ def ssh(job, index, command)
...
+old_ips="10.244.0.0/19"
+ips="10.244.0.0/16"
+gw="192.168.50.4"
+
+echo "Adding the following route entry to your local route table to enable direct container access: $ips via $gw. Your sudo password may be required." > /dev/null
+
+if [ `uname` = "Darwin" ]; then
+  sudo route delete -net $old_ips $gw
+  sudo route delete -net $ips     $gw
+  sudo route add    -net $ips     $gw
```
Comment (adamstegman) [inline spec/support/bosh.rb]: These `sudo` commands mean the test blocks waiting for user input on systems like OS X that require a password. 😞 I considered taking it out and making you run it manually (it only needs to be run once), but that would make CI harder.

File: Vagrantfile (hunk 1 above)
Comment (jeremiemv) [inline Vagrantfile]: bosh-lite wasn't using vagrant before?

Comment (adamstegman) [inline Vagrantfile]: No, the tests were using an existing AWS director before. Now they are using bosh-lite to make setup easier.

Comment (jeremiemv) [inline Vagrantfile]: Cool. Thanks for the explanation!

**Comment by ptg180** [comment]
👍

**Pull request merged**